charmcraft/main.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+
+import argparse
+import logging
+import operator
+import os
+import sys
+
+from charmcraft import logsetup
+from charmcraft.commands import version
+from charmcraft.cmdbase import CommandError
+
+logger = logging.getLogger(__name__)
+
+
+# Collect commands in different groups, for easier human consumption. Note that this is not
+# declared in each command because it's much easier to do this separation/grouping in one
+# central place and not distributed in several classes/files.
+COMMANDS_GROUPS = [
```

Comment:
Even if there are several groupS, each holding several commandS?

Reply:
yeah, because Command is an adjective at that point. You would say "groups of pens" but you would say "pen groups". There are many "colour groups" but you don't say "colours groups".

Reply:
Ah, good, thanks!!! Improved, then.